code/model/VirtualPage.php

```diff
@@ -510,8 +497,17 @@ public function getViewer($action) {
 	 * We can't load the content without an ID or record to copy it from.
 	 */
 	public function init(){
-		if(isset($this->record) && $this->record->ID){
-			if($this->record->VersionID != $this->failover->CopyContentFrom()->Version){
+		if (isset($this->record)) {
+			if (is_array($this->record)) {
```

Comment:
If your second comment is true (it never got past the first `isset()`), can we just remove this entire block of code?